Remove deprecated `Buffer` calls

![image](https://user-images.githubusercontent.com/128755/37316330-c1ada9b6-2634-11e8-977a-0b76803fa898.png)

cc @deeptibaghel

Ref mozilla/fxa#274

FWIW, I extended [**eslint-plugin-node**/recommended](http://npm.im/eslint-plugin-node) rules locally and ran it against current master (c69fee1) and got the following output:

```sh
fxa-oauth-server git:(c69fee1) $ npm run lint

> fxa-oauth-server@1.107.0 lint /Users/pdehaan/dev/github/mozilla/fxa-oauth-server/fxa-oauth-server
> eslint .


/Users/pdehaan/dev/github/mozilla/fxa-oauth-server/fxa-oauth-server/lib/browserid.js
  0:0  error  'Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla/fxa-oauth-server/fxa-oauth-server/lib/config.js
  0:0  error  'Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla/fxa-oauth-server/fxa-oauth-server/lib/routes/authorization.js
  0:0  error  'Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla/fxa-oauth-server/fxa-oauth-server/lib/routes/client/get.js
  0:0  error  'Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla/fxa-oauth-server/fxa-oauth-server/test/api.js
  0:0  error  'Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  0:0  error  'Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api
  0:0  error  'Buffer()' was deprecated since v6. Use 'Buffer.alloc()' or 'Buffer.from()' (use 'https://www.npmjs.com/package/safe-buffer' for '<4.5.0') instead  node/no-deprecated-api

/Users/pdehaan/dev/github/mozilla/fxa-oauth-server/fxa-oauth-server/test/server.js
  0:0  error  'util._extend' was deprecated since v6. Use 'Object.assign()' instead  node/no-deprecated-api
  0:0  error  'util._extend' was deprecated since v6. Use 'Object.assign()' instead  node/no-deprecated-api

✖ 9 problems (9 errors, 0 warnings)


npm ERR! Darwin 17.4.0
npm ERR! argv "/Users/pdehaan/.nvm/versions/node/v6.13.1/bin/node" "/Users/pdehaan/.nvm/versions/node/v6.13.1/bin/npm" "run" "lint"
npm ERR! node v6.13.1
npm ERR! npm  v3.10.10
```

```diff
diff --git a/.eslintrc b/.eslintrc
index 8e6a526..367f529 100644
--- a/.eslintrc
+++ b/.eslintrc
@@ -1,7 +1,18 @@
 plugins:
   - fxa
-extends: plugin:fxa/server
+  - node
+
+extends:
+  - plugin:fxa/server
+  - plugin:node/recommended
+
+root: true

 rules:
-  handle-callback-err: 0
-  semi: [2, "always"]
+  node/no-missing-require: off
+  node/no-unpublished-require: off
+  node/shebang: off
+
+  handle-callback-err: off
+  no-process-exit: off
+  semi: [error, always]
```

If each of the repos specifies something like `{ "engines": { "node": ">=6"  }}` in it's package.json file, it _may_ be possible to extend the `plugin:node/recommended` ruleset in **eslint-plugin-fxa** instead of configuring it in each repo separately, but maybe not.

--- a/docs/service-clients.md
+++ b/docs/service-clients.md
@@ -69,7 +69,7 @@ var claims = {
 };
 
 var token = base64(JSON.stringify(header)) + '.' + base64(JSON.stringify(claims));
-var sig = rsa256(Buffer(token, 'base64'), privateKey);
+var sig = rsa256(Buffer.from(token, 'base64'), privateKey);
 var jwt = token + '.' + base64(sig);
 ```
 

--- a/lib/browserid.js
+++ b/lib/browserid.js
@@ -19,7 +19,7 @@ const request = require('request').defaults({
 });
 
 function unb64(text) {
-  return Buffer(text, 'base64').toString('utf8');
+  return Buffer.from(text, 'base64').toString('utf8');
 }
 
 function Assertion(assertion) {

--- a/lib/config.js
+++ b/lib/config.js
@@ -369,7 +369,7 @@ conf.validate(options);
 conf.get('serviceClients').forEach(function(client) {
   assert(client.id, 'client id required');
   assert.equal(client.id.length, 16, 'client id must be 16 hex digits');
-  assert.equal(Buffer(client.id, 'hex').toString('hex'), client.id,
+  assert.equal(Buffer.from(client.id, 'hex').toString('hex'), client.id,
     'client id must be 16 hex digits');
   assert.equal(typeof client.name, 'string', 'client name required');
   assert.equal(typeof client.scope, 'string', 'client scope required');

--- a/lib/routes/authorization.js
+++ b/lib/routes/authorization.js
@@ -204,7 +204,7 @@ module.exports = {
         }
         return claims;
       }),
-      db.getClient(Buffer(req.payload.client_id, 'hex')).then(function(client) {
+      db.getClient(Buffer.from(req.payload.client_id, 'hex')).then(function(client) {
         logger.info('time.db_get_client', { ms: Date.now() - start });
         if (exitEarly) {
           // assertion was invalid, we can just stop here

--- a/lib/routes/client/get.js
+++ b/lib/routes/client/get.js
@@ -27,7 +27,7 @@ module.exports = {
   },
   handler: function requestInfoEndpoint(req, reply) {
     var params = req.params;
-    db.getClient(Buffer(params.client_id, 'hex')).then(function(client) {
+    db.getClient(Buffer.from(params.client_id, 'hex')).then(function(client) {
       if (! client) {
         logger.debug('notFound', { id: params.client_id });
         throw AppError.unknownClient(params.client_id);

--- a/test/api.js
+++ b/test/api.js
@@ -202,7 +202,7 @@ describe('/v1', function() {
         clientId = client.id;
         assert.equal(encrypt.hash(secret).toString('hex'), client.hashedSecret);
         assert.equal(encrypt.hash(secretPrevious).toString('hex'), client.hashedSecretPrevious);
-        badSecret = Buffer(secret, 'hex').slice();
+        badSecret = Buffer.from(secret, 'hex').slice();
         badSecret[badSecret.length - 1] ^= 1;
         badSecret = badSecret.toString('hex');
       })
@@ -1808,8 +1808,8 @@ describe('/v1', function() {
       function decodeJWT(b64) {
         var jwt = b64.split('.');
         return {
-          header: JSON.parse(Buffer(jwt[0], 'base64').toString('utf-8')),
-          claims: JSON.parse(Buffer(jwt[1], 'base64').toString('utf-8'))
+          header: JSON.parse(Buffer.from(jwt[0], 'base64').toString('utf-8')),
+          claims: JSON.parse(Buffer.from(jwt[1], 'base64').toString('utf-8'))
         };
       }
 

